File: src/main/java/com/finances/repository/IncomeRepository.java
package com.finances.repository;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;

import com.finances.entity.Income;

public interface IncomeRepository extends JpaRepository<Income, Integer> {

	Optional<List<Income>> findByDateBetween(LocalDateTime startDate, LocalDateTime endDate);

}

File: src/main/kotlin/com/finances/repository/IncomeRepository.kt
package com.finances.repository

import com.finances.entity.Income
import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDateTime

interface IncomeRepository : JpaRepository<Income, Int> {

    fun findByDateBetween(startDate: LocalDateTime, endDate: LocalDateTime): List<Income>
}
